tests/python/contrib/test_ethosu/test_codegen.py

```diff
@@ -166,8 +169,110 @@ def create_graph_activation(input_tensor_name, input_tensor_shape, input_tensor_
         infra.verify_source(compiled_models, accel_type)
 
 
+def _compare_ethosu_with_reference(
+    mod, input_data, output_data, accel_type, output_tolerance=0, print_cmm=False
+):
```

Comment:
I was wondering if maybe this and the other helper functions could go to some more common location where they would be accessible to other files as well, so that we can make use of them in other tests that follow similar pattern, namely the tests in test_lookup_table.py

Reply:
Good point, I believe this would also be useful for the tests in #9561 :)

Reply:
Agreed, I think we should move it to the 'end_to_end' infra when we add that directory.